client/src/test/java/hudson/plugins/swarm/ClientTest.java

```diff
@@ -1,16 +1,33 @@
 package hudson.plugins.swarm;
 
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.ExpectedSystemExit;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
 import static hudson.plugins.swarm.RetryBackOffStrategy.EXPONENTIAL;
 import static hudson.plugins.swarm.RetryBackOffStrategy.LINEAR;
 import static hudson.plugins.swarm.RetryBackOffStrategy.NONE;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.fail;
 
-import java.io.IOException;
-import org.junit.Test;
-
 public class ClientTest {
```

Comment:
Unfortunately, this unit test class isn't very realistic (it doesn't actually start Jenkins, for example). As such, it doesn't really test much. I think most of the tests from it currently do nothing other than printing a bunch of exceptions to standard error. I'm planning on removing this test class soon.

For a realistic test, I think we'd want to add a new test to [`plugin/src/test/java/hudson/plugins/swarm/SwarmClientIntegrationTest.java`](https://github.com/jenkinsci/swarm-plugin/blob/master/plugin/src/test/java/hudson/plugins/swarm/SwarmClientIntegrationTest.java). There, we have a Jenkins backend running, and some helper utilities to start a real instance of the Swarm Client that connects to that Jenkins backend. I think the integration test should test the two cases mentioned above:

1. Start a Swarm Client (waiting for it to connect to the backend), then kill it (ensuring the PID is left behind). Then ensure that a new Swarm Client can start successfully even though the PID file is still present (the PID should get deleted, then a new one should be created).
2. Start a Swarm Client (waiting for it to connect to the backend), then try to start a second one on top of it. The second one should fail to start because a PID file from the first should be present, and the first should still be running with the same PID.